packages/headless/src/features/insight-user-actions/insight-user-actions-state.ts

```diff
@@ -0,0 +1,75 @@
+import {InsightAPIErrorStatusResponse} from '../../api/service/insight/insight-api-client';
+
+export interface UserActionTimeline {
+  precedingSessions: UserSession[];
+  session: UserSession;
+  followingSessions: UserSession[];
+}
+
+export interface UserSession {
+  start: string;
+  end: string;
+  actions: UserActionBeta[];
+}
+
+export interface UserActionBeta {
+  actionType: UserActionType;
+  timestamp: string;
+  eventData: {
+    type: string;
+    value?: string;
+  };
+  cause?: string;
+  searchHub?: string;
+  document?: {
+    title: string;
+    clickUri: string;
+    uriHash?: string;
+    contentIdKey?: string;
+    contentIdValue?: string;
+  };
+  query?: string;
+}
+
+export enum UserActionType {
+  SEARCH = 'SEARCH',
+  CLICK = 'CLICK',
+  VIEW = 'VIEW',
+  CUSTOM = 'CUSTOM',
+}
+
+export interface UserActionsState {
+  /**
+   * The timeline of user actions.
+   */
+  timeline?: UserActionTimeline;
+  /**
+   * The names of custom actions to exclude from the user actions.
+   */
+  excludedCustomActions: string[];
+  /**
+   * The ticket creation date in the [ISO 8601](https://www.iso.org/iso-8601-date-and-time-format.html) format.
+   */
+  ticketCreationDate?: string;
+  /**
+   * The user ID to which the user's actions belong.
+   */
+  userId: string;
```

Comment:
Would that user ID relate in any way to client ID generated by relay ? 
Or is that completely different.

Could it be extracted from the search token in use by the engine in any capacity ?
If any of the above is true, should it be possible to have one ID in the search token (`foo` for example) and have userID `bar` configured for user actions ?

If this userID must be (or should be) derived from another manner, then it might not make sense to have it actually stored as a value in `user-actions-state`; but instead read from "somewhere else" when building the request.

Reply:
User id updated to be a parameter of the `fetchUserActions` action  instead of being stored in the state.